BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -91,6 +91,18 @@ public final class BlueprintView: UIView {
     /// Provides performance metrics about the duration of layouts, updates, etc.
     public weak var metricsDelegate : BlueprintViewMetricsDelegate? = nil
 
+    private var isVisible: Bool = false {
```

Comment:
I wonder if we want to add a `delegate` to allow more deeply integrating this into view controller hierarchies and their view transitions as well; eg to also allow a view controller to control visibility based on its state transitions; or to nest appearance transitions inside existing animation blocks, etc..

Reply:
> I wonder if we want to add a `delegate` to allow more deeply integrating this into view controller hierarchies and their view transitions as well; eg to also allow a view controller to control visibility based on its state transitions; or to nest appearance transitions inside existing animation blocks, etc..

I'm not 100% clear on what you have in mind, but I don't think this should be publicly writable. Every additional knob like that we add makes the internal state harder to manage.

Reply:
I was mostly thinking about: My blueprint view is hosted as the view of a view controller; it might be nice to tie "visibility" to the visibility lifecycle of that VC's viewWillAppear, etc.

Reply:
👍